### DISCUSSION TITLE
v2.2.1 (Beta)

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#3047] [Releases]
## Installation
Visit [https://getutm.app/install/][1] for the most up to date installation instructions.

## New Features (v2.2.0)
* (macOS 11.3+ Only) **Bridged networking and shared networking support**. You can configure it in the Network settings for your VM.
* (macOS 11+ and iOS 13+) **GPU acceleration for OpenGL on Linux**. Use `virtio-ramfb` or `virtio-gpu-gl-pci` display device and compatible Linux drivers (most modern Linux distros will have it already installed). Windows is not supported because there is currently no virtio-gpu driver for Windows that supports 3D acceleration.

## Changes (v2.2.1)
* GPU acceleration is now using **OpenGL** instead of **Metal** due to compatibility issues. Please report anything that works on v2.2.0 (Metal) but fails on v2.2.1 (OpenGL). If there are many cases of these, then we will add an option to switch between the two backends. From experience though, it seems like the OpenGL backend is more stable and more compatible. However, compatibility is not perfect and you may experience crashes when trying to launch an unsupported app in the guest. As before, GPU acceleration only works on Linux with the correct Mesa drivers installed for virtio.
* QEMU is updated to v6.1.0 (release)
* Fix crash when setting custom VM icon (#2387) (thanks @ktprograms)
* Disable Port Forwarding feature for bridged networking (thanks @conath)
* Fixed memory leak (#2720)

## Known Issues
* On Intel Macs, `virtio-ramfb` and `virtio-gpu-pl-pci` does not work with BIOS boot. You need to boot from UEFI which currently requires you to manually add a PC Flash device with the UEFI image. In the future, there will be an option to boot from UEFI instead of BIOS.
* On macOS 12, some UI elements are laid out incorrectly due to changes in SwiftUI apis.

Please check the full list on [Github][2] and help report any bug you find that is not listed.

[1]: https://getutm.app/install/
[2]: https://github.com/utmapp/UTM/issues

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v2.2.1'>v2.2.1 (Beta)</a>.</em>

**Comment by coreycusick** [comment]
All I can say is ever since the bugs of 2.2.0, I’ve been waiting for a fix to the OpenGL. Very happy. Will try it when I get off work on my M1 iPad Pro(iOS 15) and if I have any issues I’ll let you know. I feel Minecraft would be an interesting first test
